bpf/tests/bpf-prog-test-run/bpf-prog-test-run.go

```diff
@@ -0,0 +1,487 @@
+	if err != nil {
+		return fmt.Errorf("ct4TcpMapFromBPF failed: %v", err)
+	}
+
+	if !reflect.DeepEqual(mapAfter, mapExpected) {
+		return errors.New("resulting BPF CT map is not equal to expected")
+	}
+
+	mapDeleteAll(bpfCtMap)
+	return nil
+}
+
+func patchElf(fname string, fnameOut string) error {
+	osF, err := os.Open(fname)
+	if err != nil {
+		return err
+	}
+	defer osF.Close()
+
+	elfF, err := elf.NewFile(osF)
+	if err != nil {
+		return err
+	}
+	defer elfF.Close()
+
+	syms, err := elfF.Symbols()
+	if err != nil {
+		return err
+	}
+
+	mapSectionsOff := map[elf.SectionIndex]uint64{}
+	for secIdx, sec := range elfF.Sections {
+		if strings.HasPrefix(sec.Name, "maps") {
+			mapSectionsOff[elf.SectionIndex(secIdx)] = sec.Offset
+		}
+	}
+
+	mapSymbols := map[string]uint64{}
+	for _, sym := range syms {
+		secOff, ok := mapSectionsOff[sym.Section]
+		if !ok {
+			continue
+		}
+
+		//  This is what the current map structure looks from bpf side, so we should expect the size to be 9*4=36
+		//
+		//  struct bpf_elf_map {
+		//  	__u32 type;
+		//  	__u32 size_key;
+		//  	__u32 size_value;
+		//  	__u32 max_elem;
+		//  	__u32 flags;
+		//  	__u32 id;
+		//  	__u32 pinning;
+		//  	__u32 inner_id;
+		//  	__u32 inner_idx;
+		//  };
+		expectedMapSize := uint64(9 * 4)
+		if sym.Size != expectedMapSize {
+			log.WithFields(log.Fields{
+				"expected": expectedMapSize,
+				"actual":   sym.Size,
+			}).Fatal("invalid size")
+		}
+		mapSymbols[sym.Name] = sym.Value + secOff
+	}
+
+	outF, err := os.Create(fnameOut)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"filename": fnameOut,
+		}).Fatalf("error creating file: %v", err)
+	}
+	if err = outF.Truncate(0); err != nil {
+		log.WithFields(log.Fields{
+			"filename": fnameOut,
+		}).Fatalf("error truncating file: %v", err)
+	}
+	defer outF.Close()
+
+	if _, err = io.Copy(outF, osF); err != nil {
+		log.WithFields(log.Fields{
+			"filename": fnameOut,
+		}).Fatalf("error copying file: %v", err)
+	}
+
+	zero := [4 * 4]byte{}
+	for symName, symOff := range mapSymbols {
+		idOff := symOff + (5 * 4)
+		_, err = outF.WriteAt(zero[:], int64(idOff))
+		if err != nil {
+			log.WithFields(log.Fields{
+				"filename": fnameOut,
+			}).Fatalf("error patching file: %v", err)
+		}
+		log.WithFields(log.Fields{
+			"map": symName,
+		}).Debug("zeroed last fields")
+	}
+
+	return nil
+
+}
+
+func dumpDebugMessages(eventsReader *perf.Reader) error {
+	fmt.Printf("Log Messages:\n")
+	for {
+		record, err := eventsReader.Read()
+		if err != nil {
+			return fmt.Errorf("error reading event buffer: %v\n", err)
+		}
+
+		dm := monitor.DebugMsg{}
+		reader := bytes.NewReader(record.RawSample)
+		if err := binary.Read(reader, byteorder.Native, &dm); err != nil {
+			return fmt.Errorf("Error while parsing debug message: %v\n", err)
+		}
+
+		// special record values set from our bpf programs to mark their termination
+		if dm.SubType == monitor.DbgUnspec && dm.Arg1 == 0xe3d && dm.Arg2 == 0xe3d {
+			break
+		}
```

Comment:
Yap, Agreed.

I've seen code that already does this somewhere in our tree. I think the main issue is that we do not want to follow this approach when we integrate this test to CI tests, because we don't want to block forever (or wait for the CI timeout).

I would like to add a subsequent PR that adds some more CT tests and integrates this into the CI, so I plan to address it there.